base/iterator.jl

```diff
@@ -341,3 +341,104 @@ function collect{I<:IteratorND}(g::Generator{I})
     dest[1] = first
     return map_to!(g.f, 2, st, dest, g.iter)
 end
+
+"""
+`@shareiterators` may improve the efficiency of loops that have multiple indexes.
```

Comment:
@timholy, the warning for missing `@shareiterators` docs is because the signature needs to be

```
"""
    @shareiterators

...
"""
```

rather than an inline one.